src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1317,13 +1317,62 @@ Namespace Microsoft.CodeAnalysis.Semantics
-            Return New LazyExpressionStatement(expression, _semanticModel, syntax, type, constantValue, isImplicit)
+
+            Dim eventInvocation = TryCast(boundRaiseEventStatement.EventInvocation, BoundCall)
+
+            ' Return an invalid statement for invalid raise event statement
+            If eventInvocation Is Nothing OrElse eventInvocation.ReceiverOpt Is Nothing Then
+                Debug.Assert(boundRaiseEventStatement.HasErrors)
+                Dim children As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() ImmutableArray.Create(Of IOperation)(Create(boundRaiseEventStatement.EventInvocation)))
+                Return New LazyInvalidStatement(children, _semanticModel, syntax, type, constantValue, isImplicit)
+            End If
+
+            Dim receiver = eventInvocation.ReceiverOpt
+            Dim eventSymbol = boundRaiseEventStatement.EventSymbol
+            Dim eventReferenceSyntax = receiver.Syntax
+            Dim eventReferenceType As ITypeSymbol = eventSymbol.Type
+            Dim eventReferenceConstantValue As [Optional](Of Object) = ConvertToOptional(receiver.ConstantValueOpt)
+            ' EventReference in a raise event statement is never implicit. However, the way it is implemented, we don't get
+            ' a "BoundEventAccess" for either field backed event or custom event, and the bound nodes we get are marked as 
+            ' generated by compiler. As a result, we have to explicitly set IsImplicit to false.
+            Dim eventReferenceIsImplicit As Boolean = False
+
+            Dim boundInstance As BoundNode
+            If receiver.Kind = BoundKind.FieldAccess Then
+                ' For raising a field backed event, we will only get a field access node in bound tree.
+                Dim eventFieldAccess = DirectCast(receiver, BoundFieldAccess)
+                Debug.Assert(eventFieldAccess.FieldSymbol.AssociatedSymbol = eventSymbol)
+
+                boundInstance = eventFieldAccess.ReceiverOpt
+            Else
+                ' This is for a custom event
+                boundInstance = receiver
+            End If
+
+            Dim eventReferenceInstance As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() If(eventSymbol.IsShared, Nothing, Create(boundInstance)))
+
+            Dim eventReference As Lazy(Of IEventReferenceExpression) = New Lazy(Of IEventReferenceExpression)(Function() As IEventReferenceExpression
+                                                                                                                  Return New LazyEventReferenceExpression(eventSymbol,
+                                                                                                                                                          eventReferenceInstance,
+                                                                                                                                                          eventSymbol,
```

Comment:
>eventSymbol [](start = 154, length = 11)

@genlu Did you open an issue to follow up on the fact that we have to pass ```eventSymbol``` twice here?

Reply:
OK, let me just fix it.

Reply:
https://github.com/dotnet/roslyn/pull/22653